### ISSUE TITLE
CPU usage increased after last update after 8.2.2.26

### ISSUE THREAD

**Issue by tribunal88** [jlesage/docker-crashplan-pro#261]
Since upgrading to the latest version, 8.2.2.26, I've noticed a significant increase in CPU usage. Maybe my QNAP NAS's processor is just showing its age. Has anyone else noticed this?

**Comment by jlesage** [comment]
Do you see that CrashPlan is doing something ? After an upgrade, data is re-synchronized, which can take a lot of ressources.

**Comment by tribunal88** [comment]
Yes. It happens every time a backup is run. It uses more CPU and for a longer period of time, as backups are taking much longer. I take image copies of my VMs each week, and then back these snaps up. Prior to this release, the VMs would be finish in 4-6 hours. Now it is taking 3 to 4 days to complete. There have been no changes in VMs usage or how the underlying images might be changing themselves. It's been working as expected since 2016, both before and after I switched to the Docker CrashPlan.

**Comment by jlesage** [comment]
Ok, then something in CrashPlan itself may have changed.  I suggest you contact CrashPlan support (without mentioning you are using a Docker container) to report your issue.  Maybe they will have an explanation...

**Comment by tribunal88** [comment]
The tech says "It looks like the high CPU usage may be due to low memory allocation to the Code42 app. How much RAM does the device have?" 

The NAS has a full 8GB of memory available, and is currently using only 2.4GB. 

The resource monitor says that Crashplan is only using 1GB of memory. I'd love to give it more. IS that a setting with Crashplan? Docker? Java?

**Comment by jlesage** [comment]
To have an idea of how much RAM CrashPlan is using, you can look at `log/app.log`, under the directory you mapped to `/config`.  You should have a line like:
```
memory           = maxMemory=1536MB, totalMemory=219MB, freeMemory=146.40MB, usedMemory=72.60MB
```

You can increase the maximum amount of memory CrashPlan can use via the `CRASHPLAN_SRV_MAX_MEM` variable.  See https://github.com/jlesage/docker-crashplan-pro#environment-variables.

**Comment by jlesage** [comment]
Also, can you look at `log/service.log.0` to see if you have any errors related to the lack of memory?

**Comment by tribunal88** [comment]
I found the following line. 
memory           = maxMemory=7680MB, totalMemory=176MB, freeMemory=52.70MB, usedMemory=123.30MB

I also didn't see any errors in the service.log.0. Just a bunch of INFO entries.

**Comment by tribunal88** [comment]
This may not bode well. I just restarted the container and it uploaded some newer files from this morning (just smaller files on the NAS). However instead of going back to the large file that was only 55% complete, it is showing the entire backup as "complete" and the file previously at 55% is supposedly available for restore.

**Comment by tribunal88** [comment]
Perhaps the memory isn't expanding properly. I have the CRASHPLAN_SRV_MAX_MEM set to 7680MB. This corresponds to the Xmx java setting. Would there be a way to set the the Xms parameter? Start it off with more memory?

**Comment by jlesage** [comment]
I don't think the Xms parameter would help.  You provided 7680MB to CrashPlan but it is not using it.... Looks like the lack of memory is not the reason of the CPU usage...

**Comment by jlesage** [comment]
You can provide the `memory` line to the tech to show that memory doesn't seem to be an issue.

**Comment by eiddor** [comment]
I am seeing the same issue with CPU usage - In a steady-state, all backups completed successfully, no scanning going on, and Code42Service is hovering around 40% CPU utilization.

I've checked all of the memory stats and service.log.0 and everything appears to be fine.  @tribunal88 let me know what you find out from the tech!

**Comment by tribunal88** [comment]
Here's what the tech said "It may be having issues because you are currently backing up at least 6.3 TB on a device with only 8 GB of RAM. This could cause issues because the Code42 app needs 1 GB of RAM for every terabyte of data selected for backup."

Unfortunately, once the tech determined I was running on an "unsupported platform" based on my kernel, he or she gave me the old "CrashPlan for Small Business cannot guarantee continued functionality on unsupported operating systems."

So, unless something is found here, I'm going to have to look at other options

**Comment by eiddor** [comment]
Ugh - I imagine I'll get the same response.  I've got 8.5TB backing up from this particular container.  I'm giving it 3072M (3G) of RAM, and similar to you, it's not using much of it at all:

`memory           = maxMemory=3072MB, totalMemory=201MB, freeMemory=72.60MB, usedMemory=128.40MB`

I could try giving it 10G as a temporary test, which would mean giving another 8G to the VM itself first.  Sigh.

**Comment by tribunal88** [comment]
No dice. I reduced my backup size to under 1TB and I'm still crushing the CPU.
